providers/implementations/kdfs/sshkdf.c

```diff
@@ -159,7 +159,14 @@ static int kdf_sshkdf_set_ctx_params(void *vctx, const OSSL_PARAM params[])
 
         if (!OSSL_PARAM_get_utf8_string_ptr(p, &kdftype))
             return 0;
-        if (kdftype == NULL || kdftype[0] == '\0' || kdftype[1] != '\0')
+        /*
+         * Expect one character (byte in this case), allow for a NUL byte to
+         * follow.
+         */
+        if (kdftype == NULL
```

Comment:
This seems like an incredibly awkward pattern.
Should char be its own type?

Reply:
Ewwwwwwww...  so a separate OSSL_PARAM data type that *always* have data size 1?

Reply:
👍

Reply:
With #14168, the data size is much more reliable, and this type of hackery won't be needed.